frontend/services/entities/query_report.ts

```diff
@@ -27,13 +34,16 @@ const getSortParams = (sortOptions?: ISortOption[]) => {
 };
 
 export default {
-  load: ({ id, sortBy }: ILoadQueryReportOptions) => {
+  load: ({ id, sortBy, teamId }: ILoadQueryReportOptions) => {
     const sortParams = getSortParams(sortBy);
 
-    const queryParams = {
+    const queryParams: ILoadQueryReportQueryParams = {
       order_key: sortParams.order_key,
       order_direction: sortParams.order_direction,
     };
+    if (teamId && teamId > 0) {
+      queryParams.team_id = teamId;
+    }
```

Comment:
Is there a more idiomatic way to write this in typescript?

Reply:
Not sure it's more idiomatic, but you could write `queryParams.team_id = teamId || null` if you
want? This looks good to me too though.

Would it be good to support `team_id=0` for filtering for hosts on "No team" in the future? We can
also just make that change when we come to it.

Reply:
Just double-checked and we don't show No Team in the queries page right now, so we can tweak this when we add that ability later.

Reply:
I think the problem I encountered was the `currentTeamId` variable was getting set to `-1` when there was no team selected. This is why I couldn't just do `teamId || null`.

Reply:
This happens when you load a global query detail page.
<img width="1485" alt="Screenshot 2024-12-17 at 3 02 03 PM" src="https://github.com/user-attachments/assets/f6635e75-2fba-46ab-8fdb-d5ad57b45c08" />

Reply:
Ah